packages/core/src/schema/schema.ts

```diff
@@ -79,8 +83,21 @@ type GetTable<
         [columnName: string]: Column;
       };
 
-export const createTable = <const table>(t: GetTable<table>): table =>
-  t as table;
+type GetConstraints<
+  constraints,
+  table,
+  ///
+  columnName extends string = ExtractNonVirtualColumnNames<table>,
+> = {} extends constraints
+  ? {}
+  : {
+      [name in keyof constraints]: Index<columnName | readonly columnName[]>;
+    };
+
+export const createTable = <const table, const constraints>(
+  t: GetTable<table>,
+  c?: GetConstraints<constraints, table>,
+): readonly [table, constraints] => [t as table, c as constraints];
```

Comment:
I think I'd find this easier to follow as `{ columns, constraints }`, but open if there's a reason not to do that.